tests/unit/bokeh/core/property/test_descriptors.py

```diff
@@ -259,17 +267,71 @@ class Foo:
         assert d.__doc__ == f.__doc__
         assert d.units_prop == g
 
-class Test_AliasSpecDescriptor:
+class Test_AliasDescriptor:
     def test___init__(self) -> None:
-        class Foo:
-            '''doc'''
-            pass
-        f = Foo()
-        d = bcpd.AliasPropertyDescriptor("foo", "bar", f)
+        f = Alias("bar")
+        d = bcpd.AliasPropertyDescriptor("foo", f)
+        assert d.name == "foo"
+        assert d.aliased_name == "bar"
+        assert d.property == f
+        assert d.__doc__ == "This is a compatibility alias for the 'bar' property."
+
+    def test_values(self) -> None:
+        class Some(Model):
+            p0 = Int(default=17)
+            p1 = Alias("p0")
+
+        obj = Some()
+
+        assert obj.p0 == 17
+        assert obj.p1 == 17
+
+        obj.p0 = 18
+        assert obj.p0 == 18
+        assert obj.p1 == 18
+
+        obj.p1 = 19
+        assert obj.p0 == 19
+        assert obj.p1 == 19
+
+class Test_DeprecatedAliasDescriptor:
+    def test___init__(self) -> None:
+        f = DeprecatedAlias("bar", since=(3, 1, 0))
+        d = bcpd.DeprecatedAliasPropertyDescriptor("foo", f)
         assert d.name == "foo"
         assert d.aliased_name == "bar"
         assert d.property == f
-        assert d.__doc__ == "This is a compatibility alias for the ``bar`` property"
+        assert d.__doc__ == """\
+This is a backwards compatibility alias for the 'bar' property.
+
+.. note::
+    Property 'foo' was deprecated in bokeh 3.1.0 and will be removed
+    at some point. Update your code to use 'bar' instead.
+"""
+
+    @patch("warnings.warn")
+    def test_warns(self, mock_warn: MagicMock) -> None:
+        class Some(Model):
+            p0 = Int(default=17)
+            p1 = DeprecatedAlias("p0", since=(3, 1, 0))
+
+        obj = Some()
+
+        assert obj.p0 == 17
+        assert not mock_warn.called
+
+        assert obj.p1 == 17
+        assert mock_warn.called
+        assert mock_warn.call_args[0] == ("p1 was deprecated in Bokeh 3.1.0 and will be removed, use p0 instead.", BokehDeprecationWarning)
```

Comment:
FYI there are some newer, nicer (IMO) APIs for this these days
```
from unittest.mock import call
assert mock_open.has_call(call("foo", "r", newline="", encoding="utf8"))
```

Reply:
`has_call` doesn't seem like a supported API, but I upgraded this code to an `assert_*` variant.

Reply:
Weird, I guess this is though:

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls

I was mostly referring to `call(...)` which is nice for packing up function calls.